app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt

```diff
@@ -218,182 +228,294 @@ private fun MessageComposer(
+                    keyboardHeightOffSet,
+                    messageComposerState,
+                    onSendAttachment,
+                    onMessageComposerError,
+                    isFileSharingEnabled,
+                    tempCachePath
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun AttachmentOptions(
+    keyboardHeightOffSet: KeyboardHeight,
+    messageComposerState: MessageComposerInnerState,
+    onSendAttachment: (AttachmentBundle?) -> Unit,
+    onMessageComposerError: (ConversationSnackbarMessages) -> Unit,
+    isFileSharingEnabled: Boolean,
+    tempCachePath: Path
+) {
+    Box(
+        Modifier
+            .fillMaxWidth()
+            .height(keyboardHeightOffSet.height)
+            .absoluteOffset(y = messageComposerState.fullScreenHeight - keyboardHeightOffSet.height)
+    ) {
+        Divider()
+        AttachmentOptionsComponent(
+            messageComposerState.attachmentInnerState,
+            onSendAttachment,
+            onMessageComposerError,
+            isFileSharingEnabled,
+            tempCachePath,
+            Modifier.align(Alignment.Center)
+        )
+    }
+}
+
+@ExperimentalAnimationApi
+@Composable
+private fun MessageComposeActionsBox(
+    modifier: Modifier,
+    transition: Transition<MessageComposeInputState>,
+    messageComposerState: MessageComposerInnerState,
+    focusManager: FocusManager
+) {
+    Column(
+        modifier
+            .wrapContentSize()
+    ) {
+        Divider()
+        Box(Modifier.wrapContentSize()) {
+            transition.AnimatedVisibility(
+                visible = { messageComposerState.messageComposeInputState != MessageComposeInputState.Enabled },
+                // we are animating the exit, so that the MessageComposeActions go down
+                exit = slideOutVertically(
+                    targetOffsetY = { fullHeight -> fullHeight / 2 }
+                ) + fadeOut()
+            ) {
+                MessageComposeActions(messageComposerState, focusManager)
+            }
+        }
+    }
+}
+
+@ExperimentalAnimationApi
+@Composable
+private fun SendActions(
+    modifier: Modifier,
+    messageComposerState: MessageComposerInnerState,
+    transition: Transition<MessageComposeInputState>,
+    onSendButtonClicked: () -> Unit
+) {
+    Box(modifier) {
+        Row(Modifier.padding(end = dimensions().spacing8x)) {
+            if (messageComposerState.sendButtonEnabled) {
+                ScheduleMessageButton()
+            }
+            transition.AnimatedVisibility(
+                visible = { messageComposerState.messageComposeInputState != MessageComposeInputState.Enabled },
+                enter = fadeIn(),
+                exit = fadeOut()
+            ) {
+                SendButton(
+                    isEnabled = messageComposerState.sendButtonEnabled,
+                    onSendButtonClicked = onSendButtonClicked
+                )
+            }
+        }
+    }
+}
```

Comment:
Same than above. Can we move to a different file? Maybe one grouping all the MessageComposeActions ?